veascan-web/src/App.tsx

```diff
@@ -24,13 +25,25 @@ const App = () => {
   const [currentPage, setCurrentPage] = useState(1);
   const [
     pageTracking,
-    { push: pushPageTracking, removeAt: removeAtPageTracking },
+    {
+      push: pushPageTracking,
+      removeAt: removeAtPageTracking,
+      clear: clearPageTracking,
+    },
   ] = useList<IPageTracking>();
   const { data } = useSnapshots(
     pageTracking.at(-1)?.shownSnapshots,
     pageTracking.at(-1)?.timestamp,
     SNAPSHOTS_PER_PAGE
   );
+
+  const { fromChain, toChain, statusFilter } = useFiltersContext();
+
+  useEffect(() => {
+    clearPageTracking();
```

Comment:
Can we add a conditional here? like:
```typescript
if (currentPage !== 1) {
  clear
  set page to 1
}
```
without the if statement the state will be updated during the first render and that will cause a second render.

without the if (3 renders):
![image](https://github.com/kleros/vea/assets/80388279/566eaab0-eae7-4255-b97c-a0e836496008)

with the if (2 renders):
![image](https://github.com/kleros/vea/assets/80388279/ae5e660e-2700-4227-9679-a7e014a0b0f9)